change/@ni-nimble-components-aca919ed-c567-4f7b-8c65-7310408a45b2.json

```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Use anchored region in Select and Combobox",
+  "packageName": "@ni/nimble-components",
```

Comment:
> I initially looked at using [Floating-UI](https://floating-ui.com/), as that is what FAST 2.0 is using to solve this problem. (see https://github.com/ni/nimble/pull/1234) However, for consistency, that would require adding support to both the Select/Combobox as well as updating the menu button, the menu item (for submenu support), and the tooltip. Each case was a significant change, and I ran into several problems, including with the action menu in the table. It turned into a 40+ file change with a lot of risk. I sunk more time into it than I wanted, and still didn't feel close to done.

As discussed in stand-up this trade-off can make sense for today but is there anything we need to capture now to help our future selves with the idea that we expect to migrate to floating ui probably within 6 months or so?

For example:
- Are there issues we found copying the new select pattern based on floating UI that we would expect in the upstream fast-foundation select we should report now?
- Are there issues we need to report to floating-ui about behaviors we saw?
- Are there prototype branches we should reference in the future and capture?
- Should we make an HLD / research spike to evaluate what the transition to floating-ui looks like?

Reply:
I'm not convinced that I ran into any bugs in FAST code or in Floating-UI. It was more that I saw behaviors I didn't expect or understand, and could not come up with satisfying resolutions in the time I had. I also had to do some ugly things like override private class members, which should not be the case when we ultimately update our version of FAST. There was also risk around what existing code would continue to work properly as I tried to integrate it with the Floating-UI code.

I don't expect we will want to move to Floating-UI before updating our version of FAST. I don't see a reason to do so. If we wait until updating FAST, then it will be doing the work of integrating Floating-UI for us. From that perspective, there's no need to make a plan/HLD for doing the transition to Floating-UI ourselves.

Reply:
All right, sounds like the argument is it'll be trivial to migrate our forked stuff / completely custom stuff to floating ui once the new fast foundation components are released and we don't need any pre-work to make that transition smoother. I'll hold you to it!